prediction_market_agent/agents/safe_guard_agent/guards/hash_checker.py

```diff
@@ -0,0 +1,30 @@
+from safe_eth.safe import SafeTx
+
+from prediction_market_agent.agents.safe_guard_agent.safe_api_models.detailed_transaction_info import (
+    DetailedTransactionResponse,
+)
+from prediction_market_agent.agents.safe_guard_agent.validation_result import (
+    ValidationResult,
+)
+
+
+def validate_safe_transaction_hash(
+    new_transaction: DetailedTransactionResponse,
+    new_transaction_safetx: SafeTx,
+    history: list[DetailedTransactionResponse],
+) -> ValidationResult:
+    """Function that checks if the safe_tx_hashes match."""
+
+    if not new_transaction.detailedExecutionInfo:
```

Comment:
Here, you can use `check_not_none` to get rid of mypy problems, because without detailedExecutionInfo, it should error out much sooner anyway: https://github.com/gnosis/prediction-market-agent/blob/e77d463430d9142b28b18883e5a685fab31a677b/prediction_market_agent/agents/safe_guard_agent/safe_api_utils.py#L104